### DISCUSSION TITLE
Can't figure out how to cull low-scoring skeletons

### DISCUSSION THREAD

**Discussion by olinesn** [talmolab/sleap#991] [Help!] [answered]
Hello,

I've created a rich dataset where I'd like to cull low-scoring skeletons (below 0.7) during inference before tracking. As an example, if I have a video with two mice, and SLEAP provides two skeletons scoring 0.4 and 0.9, I'd like the 0.4 skeleton to be deleted before tracking.

Perhaps I'm doing something wrong, but I can't figure out how to delete these skeletons aside from manually in the GUI ('Labels > Delete All Predictions with Low Score'). Here's my syntax:
`sleap-track --verbosity rich --tracking.tracker flow --tracking.target_instance_count 2 --tracking.pre_cull_to_target 2 --tracking.pre_cull_iou_threshold .9 --tracking.post_connect_single_breaks 1 --tracking.clean_instance_count 2 --tracking.clean_iou_threshold .9  --tracking.similarity instance  --tracking.track_window 10 --model "..." --model "..." "....mp4"`

The result includes two tracks, which both have skeletons below 0.9. Is it possible that the above syntax culls skeletons until there are only two left, and then leaves them regardless of their score?

Thanks!

**Comment by roomrys** [comment, marked as answer]
Hi @olinesn,

There is an indirect way of filtering instances with a low score using the [`--peak_threshold`](https://sleap.ai/guides/cli.html?highlight=cli#sleap-track) argument (default: 0.2). This will determine whether to consider a proposed body part location valid based on the _node_ score (not the score of the entire instance - which is the average of all the node scores). Setting a higher `peak_threshold` will yield less predicted nodes, but the nodes that are returned will have a higher score. We should just have a command line option to delete the instances based on score (i.e. what the user sees). 

The `pre_cull_iou_threshold` and `clean_iou_threshold` arguments cull instances based on the [intersection over union](https://learnopencv.com/intersection-over-union-iou-in-object-detection-and-segmentation/) of the instances' bounding boxes (not their scores). Also, if you `pre_cull_iou_threshold` before tracking, then you do not need to `clean_iou_threshold` after tracking as the instances that meet this criteria will have already been culled.

Let us know if setting the `peak_threshold` helps (although it beats around the bush a bit).

Thanks,
Liezl

**Reply by olinesn** [reply]
Hi Liezl,

This is huge, I prefer your suggestion to my proposed solution, that way the good nodes from the bad skeletons stay, and I don't need a skeleton threshold at all!

Thanks also for the tip about the cleanup step, good to know it's redundant.
